src/nvim/terminal.c

```diff
@@ -293,11 +293,11 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
   vterm_get_size(term->vt, &curheight, &curwidth);
 
   if (!width) {
-    width = (uint16_t)curwidth;
+    width = UINT16_MAX;
   }
 
   if (!height) {
-    height = (uint16_t)curheight;
+    height = UINT16_MAX;
```

Comment:
Maybe a better strategy would be to remove the `width` and `height` parameters and set them to `UINT16_MAX` unconditionally. In the four call sites of `terminal_resize()`, two pass `0` and `0`, and the other two pass the height or width of the window being resized. However, even if we know the height of *a* window being resized, we must still check every window, so it doesn't really save us anything.